Add custom headers and footers to the generated document

chasenlehara opened the pull request:
Previously, a template’s custom header and footer were not part of the generated HTML preview at all.

Now, the headers and footers are included in the SSR page when the `renderTemplateFootersAndHeaders` attribute is `true`. This will allow us to conditionally turn on the headers and footers when DAT renders the HTML preview but not when it’s generating the PDF.

This adds a number of attributes to one of the existing templates because they are required for the template to successfully render; see this issue for more info: https://github.com/CCALI/a2jdeps/issues/55

Part of https://github.com/CCALI/a2jdeps/issues/53

mitch-mitchel commented on elements/a2j-template-ssr/demo.html:
```diff
@@ -17,10 +17,11 @@
       var stache = require('can-stache/');
 
-      require('styles.less!');
+      // require('styles.less!');
```
should this just be deleted?

mitch-mitchel commented on models/tests/a2j-template-test.js:
```diff
@@ -22,7 +22,7 @@ describe('A2JTemplate Model', function () {
 
     return promise.then(function (a2jTemplates) {
-      assert.equal(a2jTemplates.length, 3, 'should have 3 total templates')
+      assert.equal(a2jTemplates.length, 4, 'should have 3 total templates')
```
missed updating the 'should' message to 4

chasenlehara replied:
Good catches, thank you!

Commit:
Add custom headers and footers to the generated document

Previously, a template’s custom header and footer were not part of the generated HTML preview at all.

Now, the headers and footers are included in the SSR page when the `renderTemplateFootersAndHeaders` attribute is `true`. This will allow us to conditionally turn on the headers and footers when DAT renders the HTML preview but not when it’s generating the PDF.

This adds a number of attributes to one of the existing templates because they are required for the template to successfully render; see this issue for more info: https://github.com/CCALI/a2jdeps/issues/55

Part of https://github.com/CCALI/a2jdeps/issues/53
